arrow-cast/src/cast/mod.rs

```diff
@@ -9980,7 +9980,26 @@ mod tests {
         };
         let result = cast_with_options(&array, &output_type, &options);
         assert_eq!(result.unwrap_err().to_string(),
-                   "Invalid argument error: 123456790 is too large to store in a Decimal128 of precision 6. Max is 999999");
+                   "Invalid argument error: 123456789 is too large to store in a Decimal128 of precision 6. Max is 999999");
+    }
+
+    #[test]
+    fn test_decimal_to_decimal() {
```

Comment:
might wanna give this a better name than what I came up with :D 

```suggestion
    fn test_decimal_to_decimal_same_scale() {
```
maybe?